toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -383,6 +390,42 @@ func initializeRpmDatabase(installRoot string) (err error) {
 	return
 }
 
+// TdnfInstall installs a package in the current environment
+func TdnfInstall(packageName, installRoot string, currentPackagesInstalled, totalPackages int) (packagesInstalled int, err error) {
```

Comment:
I don't think we want to expose this function. It uses `ReportAction` and `ReportPercentComplete`. These calls are special for the ISO installers and are expected to run at an exact point in the ISO's lifetime. If run by other code while the progress emitter is turned on (`EnableEmittingProgress()`) it may cause undesired visual artifacts in the installer or noise in the logs.

Reply:
I've made the reporting behaviour optional

---
In reply to: [551640363](https://github.com/microsoft/CBL-Mariner/pull/481#discussion_r551640363) [](ancestors = 551640363)